### PR TITLE
Add RTL generation for modules and some rtl fixtures

### PR DESCRIPTION
**Pull request by Danoosh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | Add RTL generation for modules and some rtl fixtures
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | [BOOM-4319](http://forge.prestashop.com/browse/BOOM-4319)
| How to test?  | Install Prestashop by using an rtl language or add an rtl language to active shop. install a new module that has css files and open configuration to see the new rtl button.

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8691)

### CODE REVIEW

File: classes/Language.php [hunk 1]
```diff
@@ -188,7 +188,11 @@ public function add($autodate = true, $nullValues = false, $only_add = false)
         }
 
         if ($this->is_rtl) {
-            self::installRtlStylesheets(true, false, null, null, (defined('PS_INSTALLATION_IN_PROGRESS') ? true : false));
+            self::getRtlStylesheetProcessor()
+                ->setIsInstall((defined('PS_INSTALLATION_IN_PROGRESS') ? true : false))
```
Comment (Quetzacoalt91) [inline classes/Language.php]: You can remove the ternary test as you will get the boolean you're looking for.

File: classes/Language.php [hunk 2]
```diff
@@ -1338,54 +1346,31 @@ public static function updateMultilangFromClass($table, $className, $lang)
...
 
-            $generator->generateInDirectory($frontDir.($themeName?$themeName:'classic'));
-        }
+        $processor = new RtlStylesheetProcessor(
+            $adminDir,
+            $themesDir,
+            array(
+                _PS_MODULE_DIR_.'gamification',
+                _PS_MODULE_DIR_.'welcome',
+                _PS_MODULE_DIR_.'cronjob',
```
Comment (Quetzacoalt91) [inline classes/Language.php]: I'm quite worried by this list. It seems to be lost in the middle of the code, and we do not know why especially these modules should be concerned. What if another native/contributor module want to be added here?

File: classes/Language.php (hunk 1 above)
Comment (eternoendless) [reply classes/Language.php]: True, I'll change that

File: classes/Language.php (hunk 2 above)
Comment (eternoendless) [reply classes/Language.php]: I agree. It's basically a whitelist of native modules that should be transformed to RTL automatically. I thought about setting it as a class property, but then there was the problem of the concatenation, availability of the constant at construction, and so on.

Originally, the idea was to transform _all_ modules. However, that could cause problems on modules that already were RTL-compatible, so this shortlist was made.

Modules can also be converted to RTL manually using a new button that appears on the module's configuration page (but they need to have one).

I think that eventually modules should be able to set up a property that specifies if the module should be converted to RTL automatically, manually, or not converted to RTL at all.

I don't believe we need to settle that right away though.

Comment (Quetzacoalt91) [reply classes/Language.php]: If you need something more complex (i.e retuning an answer that depend on something), we can use the hook system for that.
Only the module registered to that hook AND specifying RTL compliant would be taken in account. :)

Comment (eternoendless) [reply classes/Language.php]: Yeah, I think we can set that up later :)

File: controllers/admin/AdminThemesController.php [hunk 3]
```diff
@@ -326,7 +326,10 @@ public function postProcess()
                 $this->redirect_after = $this->context->link->getAdminLink('AdminThemes');
             }
         } elseif (Tools::isSubmit('submitGenerateRTL') && Tools::getValue('PS_GENERATE_RTL')) {
-            Language::installRtlStylesheets(false, true, Tools::getValue('PS_THEMES_LIST'));
+            Language::getRtlStylesheetProcessor()
+                ->setProcessFOThemes(array(Tools::getValue('PS_THEMES_LIST')))
+                ->setRegenerate(true)
+                ->process();
             $this->confirmations[] = $this->trans(
                 'Your RTL stylesheets has been generated successfully',
```
Comment (LouiseBonnard) [inline controllers/admin/AdminThemesController.php:334]: Domains : Admin/Design/Notification

Comment (Quetzacoalt91) [reply controllers/admin/AdminThemesController.php:334]: This domain is already used here :)

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @Danoosh

File: classes/Language.php (hunk 2 above)
Comment (Danoosh) [reply classes/Language.php]: I think a property in module is enough. we can set a property to each module.
We can use $this->use_auto_rtl = true /false in module construction method,
then we can rtl it during Prestashop or module installation and when an RTL language is added.

**Pull request merged**